### ISSUE TITLE
Unsupervised learning on TU dataset.

### ISSUE THREAD

**Issue by jerryzhang1119** [susheels/adgcl#4]
Hi!
I'm trying to reproduce your AD-GCL results on TU-dataset (unsupervised learning). I could achieve nearly the same results as your paper reported. However, when I split the training process into two steps (AD_GCL for latent vector generation and using latent vector for linear classification (linear SVC). The result is bad enough (training : 65%, val and test: ~50%)). I wonder what is the difference between these two training strategies.
Thanks a lot

**Comment by susheels** [comment]
Thanks for looking into our work. 

So the protocol we use for unsupervised learning is that during training, model (GNN) is only trained using self-supervised AD-GCL. During evaluation we use the frozen GNN and train a linear classifier on top.  These are the reported values.

In your case, it is possible to train both AD-GCL and linear head together as long as you introduce a stop gradient right before the linear head to match the unsupervised protocol. Did you try this way ?

Thanks,
Susheel

**Comment by jerryzhang1119** [comment]
Thanks for your reply~
Yes. I have tried this way.  I would see the latent vector distribution, such as TSNE for visualization.(I thought this training strategy is nearly the same as your code). After self-supervised learning (AD-GCL), which doesn't introduce the label information. And I could obtain the latent vector (embeded dim x 1) for each graph. Then I train a linear SVC for the downstream task(supervised). But the performance is not good enough.

**Comment by susheels** [comment]
Hi, it would be great if you could be a little more specific. Which dataset? did you train linear svc at different epochs of AD-GCL training? What were the hyperparameters?  Like for e.g., TU dataset we do k-fold evaluation. Did you do that ?

**Comment by jerryzhang1119** [comment]
I have found the problem, I split the code into two stages and load the optimal model to generate latent vector for SVC. But actually the weight of AD-GCL changes as the SVC training. Is my expression correct? I have save the latent vector and then load the latent vector, then training SVC. The performance is getting close to your report. Thank you for your kind suggestion and your work is excellent~
